src/amo/components/AddonReviewManager/index.js

```diff
@@ -0,0 +1,144 @@
+/* @flow */
+import makeClassName from 'classnames';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import {
+  SAVED_RATING,
+  STARTED_SAVE_RATING,
+  STARTED_SAVE_REVIEW,
+  updateAddonReview,
+} from 'amo/actions/reviews';
+import { withFixedErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import { sanitizeHTML } from 'core/utils';
+import DismissibleTextForm from 'ui/components/DismissibleTextForm';
+import Rating from 'ui/components/Rating';
+import type { AppState } from 'amo/store';
+import type { DispatchFunc } from 'core/types/redux';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { FlashMessageType, UserReviewType } from 'amo/actions/reviews';
+import type { I18nType } from 'core/types/i18n';
+import type { OnSubmitParams } from 'ui/components/DismissibleTextForm';
+
+import './styles.scss';
+
+type Props = {|
+  review: UserReviewType,
+|};
+
+type InternalProps = {|
+  ...Props,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  flashMessage?: FlashMessageType | void,
+|};
+
+export class AddonReviewManagerBase extends React.Component<InternalProps> {
+  onSubmitRating = (rating: number) => {
+    const { errorHandler, dispatch, review } = this.props;
+
+    dispatch(
```

Comment:
should we have invariant checks here?

Reply:
It's not necessary because all of these properties cannot be empty (see [InternalProps](https://github.com/mozilla/addons-frontend/pull/5997/files/0887b8484e3fc3f4de326e36109d599fbb9dcebf#diff-dddd6fbd1ffe3a0900aa9b0da5db5fe8R31)). If a caller leaves them empty, Flow will show an error about it. Thus, we don't need any extra `invariant()` calls.

Reply:
ok, sorry for dupe questions :/ I guess I still have to sort this out in my mind a bit..I think b/c I remember adding them in some other scenarios. Maybe render..